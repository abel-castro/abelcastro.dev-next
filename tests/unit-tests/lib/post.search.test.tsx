import { render } from '@testing-library/react';
import { usePathname } from 'next/navigation';
import { describe, expect, test, vi } from 'vitest';

import Search, {
    getSearchParams,
} from '../../../app/components/posts/post-search';

vi.mock('next/navigation', () => ({
    usePathname: vi.fn(),
    useRouter: vi.fn(() => ({
        replace: vi.fn(),
    })),
    useSearchParams: vi.fn(() => new URLSearchParams()),
}));

describe('Search Component', () => {
    test('should render the search component matching the snapshot', () => {
        const { container } = render(<Search placeholder="Search posts" />);
        expect(container).toMatchSnapshot();
    });

    test('should call usePathname', () => {
        render(<Search placeholder="Search posts" />);
        expect(usePathname).toHaveBeenCalled();
    });
});

describe('getSearchParams', () => {
    test('should set page to 1 and add query parameter if term is provided', () => {
        const searchParams = new URLSearchParams();
        const term = 'test';
        const result = getSearchParams(searchParams, term);
        expect(result.get('page')).toBe('1');
        expect(result.get('query')).toBe(term);
    });

    test('should set page to 1 and remove query parameter if term is empty', () => {
        const searchParams = new URLSearchParams({ query: 'test' });
        const term = '';
        const result = getSearchParams(searchParams, term);
        expect(result.get('page')).toBe('1');
        expect(result.has('query')).toBe(false);
    });

    test('should preserve other existing parameters', () => {
        const searchParams = new URLSearchParams({ other: 'value' });
        const term = 'test';
        const result = getSearchParams(searchParams, term);
        expect(result.get('page')).toBe('1');
        expect(result.get('query')).toBe(term);
        expect(result.get('other')).toBe('value');
    });
});
