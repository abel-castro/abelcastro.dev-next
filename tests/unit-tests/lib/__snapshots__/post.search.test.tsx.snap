// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Search Component > should render the search component matching the snapshot 1`] = `
<div>
  <form
    class="relative flex mt-4 sm:mt-0 w-full sm:w-auto sm:flex justify-center"
    data-testid="search-form"
  >
    <label
      class="sr-only"
      for="search"
    >
      Search
    </label>
    <input
      class="peer block rounded border border-gray-200 py-[9px] lg:pl-8 pl-4 text-sm outline-2 text-gray-500 w-full max-w-xs"
      placeholder="Search posts"
    />
    <svg
      aria-hidden="true"
      class="hidden lg:block absolute left-3 top-1/2 h-[18px] w-[18px] -translate-y-1/2 text-gray-500 peer-focus:text-gray-900 "
      data-slot="icon"
      fill="none"
      stroke="currentColor"
      stroke-width="1.5"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m21 21-5.197-5.197m0 0A7.5 7.5 0 1 0 5.196 5.196a7.5 7.5 0 0 0 10.607 10.607Z"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
    </svg>
  </form>
</div>
`;
